### DISCUSSION TITLE
Global parameters config/stage file

### DISCUSSION THREAD

**Discussion by mstrakow** [Azure-Player/azure.datafactory.tools#314] [Show and tell]
During deployment I was trying to replacing all global variables environment-related. I'm using one .json config file for all env.
It looks like below :
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/09ede8f1-476e-4930-a276-eda281ad2ee2)

what I'm getting during process is :
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/1b2c5b3c-a493-4d60-8e72-4279cf1e27a8)

any clue how to properly configure .json file format to be able to replace global variables ?
Thanks in advance for any hint.

**Comment by NowinskiK** [comment]
List all env variables available for the agent to make sure you use the correct names.
You shouldn't use `-` in variable names - use `_` instead.
In pipelines, you can reference predefined variables as environment variables. 
For example, the variable `Build.ArtifactStagingDirectory` becomes the variable `BUILD_ARTIFACTSTAGINGDIRECTORY`.
https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

**Reply by mstrakow** [reply]
Thank You for Your feedback Kamil. 
I don't think this is a main cause of what is happening.
I've created same configuration file but in .csv with the same naming convention of env. variables
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/fb5e2128-a200-4a50-a169-c413bdf16ad1)

and it throws 
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/1c35ea43-87a0-4676-a4de-b21b83496f09)

PROJECT-NAME is a variable that I'm passing as pipeline parameter and I see this variable in env. 
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/2918968c-6d27-47f0-bc39-1e0c9316b616)

Process is not complaining about 3 previous variables that are in config file DBX-DOMAIN,DBX-ID,DBX-CLUSTER-ID,
which are also visible on env. variables.
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/7e9efbd7-b16e-4358-b05f-be56d5e5b887)

When I switch to .json config file I still get error like in my 1st comment, but maybe this is because whole .json is checked at once and if one of env. is missing is giving ##[error]Invalid JSON primitive: .
I don't know this make sense.

**Reply by NowinskiK** [reply]
Did you try:
1) having only one item in config file, then
2) replace variable name from `PROJECT-NAME` to `PROJECT_NAME`?

How do you think why Microsoft does NOT use dashes in environment names? 
Well, if you don't want to follow my suggestions then carry on trial & error...

**Reply by mstrakow** [reply]
Thanks Kamil for Your response. 
You right that "-" are not supported by Azure, but "-" is not replaced by default to "_",  that I was confused. But anyway, I removed all "-" from my variables and with .csv param file  it worked as expected. 
But not with .json it's still : 
##[error]Invalid JSON primitive: .

I even try with single parameter in .json like 
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/5d8183d8-3845-4922-bb0b-ea36f37b5afa)

![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/33649613/d61db254-043a-4e65-880b-4f5a6ac3b84e)
